http.go

```diff
@@ -138,19 +138,45 @@ func (p *HTTPPool) PickPeer(key string) (ProtoGetter, bool) {
 	return nil, false
 }
 
-func (p *HTTPPool) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	// Parse request.
+func (p *HTTPPool) parseRequest(r *http.Request) (string, string, bool) {
```

Comment:
Use named result parameters here, otherwise (string, string, bool) is super confusing, especially without docs.

Reply:
@bradfitz Modified. Thanks for your review.